test-suite/src/function/cast.rs

```diff
@@ -74,6 +83,105 @@ test_case!(cast_literal, async move {
             r#"SELECT CAST(NULL AS FLOAT) AS cast FROM Item"#,
             Ok(select_with_null!(cast; Null)),
         ),
+        (
+            r#"SELECT CAST(true AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(1,0))),
+        ),
+        (
+            r#"SELECT CAST(false AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(0,0))),
+        ),
+        (
+            r#"SELECT CAST(number AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(1,0))),
+        ),
+        (
+            r#"SELECT CAST("1.1" AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(11,1))),
+        ),
+        (
+            r#"SELECT CAST(1 AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(10, 1))),
+        ),
+        (
+            r#"SELECT CAST(-1 AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(-10, 1))),
+        ),
+        (
+            r#"SELECT CAST("foo" AS Decimal) AS cast FROM Item"#,
+            Err(ValueError::LiteralCastFromTextToDecimalFailed("foo".to_owned()).into()),
+        ),
+        (
+            r#"SELECT CAST(NULL AS Decimal) AS cast FROM Item"#,
+            Ok(select_with_null!(cast; Null)),
+        ),
+        (
+            r#"SELECT CAST(true AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(1,0))),
+        ),
+        (
+            r#"SELECT CAST(false AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(0,0))),
+        ),
+        (
+            r#"SELECT CAST(number AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(1,0))),
+        ),
+        (
+            r#"SELECT CAST("1.1" AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(11,1))),
+        ),
+        (
+            r#"SELECT CAST(1 AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(10, 1))),
+        ),
+        (
+            r#"SELECT CAST(-1 AS decimal) AS cast FROM Item"#,
+            Ok(select!(cast Decimal; Dec::new(-10, 1))),
+        ),
+        (
+            r#"SELECT CAST("foo" AS Decimal) AS cast FROM Item"#,
+            Err(ValueError::LiteralCastFromTextToDecimalFailed("foo".to_owned()).into()),
+        ),
+        (
+            r#"SELECT CAST(NULL AS Decimal) AS cast FROM Item"#,
+            Ok(select_with_null!(cast; Null)),
+        ),
+        (
+            r#"SELECT CAST(mytext AS Decimal) AS cast FROM test"#,
+            Err(ValueError::ImpossibleCast.into()),
+            //Err(ValueError::LiteralCastFromTextToDecimalFailed("foo".to_owned()).into()),
```

Comment:
Sure, I will make the change from Dec to Decimal.  I chose Dec because I thought there was a conflict with Value::Decimal, but I will double check.

Reply:
Using Decimal instead of Dec seems to work ok, so I made the change.  Let me know if anything else needs to be changed.

I believe the code is now up to date with your suggestions.  If not, let me know